zio-http-logging/src/main/scala-2/zhttp/logging/macros/LoggerMacroImpl.scala

```diff
@@ -0,0 +1,59 @@
+package zhttp.logging.macros
+
+import zhttp.logging.{LogLevel, Logger}
+
+import scala.reflect.macros.whitebox
+
+/**
+ * Macro inspired from log4s.
+ */
+private[zhttp] object LoggerMacroImpl {
+
+  /** A macro context that represents a method call on a Logger instance. */
+  private[this] type LogCtx = whitebox.Context { type PrefixType = Logger }
+
+  /**
+   * Log a message reflectively at a given level.
+   */
+  private[this] def reflectiveLog(
+    c: LogCtx,
+  )(msg: c.Expr[String], error: Option[c.Expr[Throwable]])(logLevel: LogLevel) = {
+    import c.universe._
+    type Tree = c.universe.Tree
+
+    val cname: Tree          = q"${c.internal.enclosingOwner.owner.fullName}"
+    val lno: Tree            = q"${c.enclosingPosition.line}"
+    val sourceLocation: Tree =
+      if (logLevel == LogLevel.Trace)
+        q"Some(_root_.zhttp.logging.Logger.SourcePos($cname, $lno))"
+      else
+        q"None"
+    val level: Tree          = q"_root_.zhttp.logging.LogLevel.${TermName(logLevel.name.toLowerCase.capitalize)}"
+    val isEnabled: Tree      = q"${c.prefix.tree}.isEnabled(${level})"
```

Comment:
```suggestion
    val isEnabled: Tree      = q"${c.prefix.tree}.is${level}Enabled"
```

I think this will perform a bit better.